# Analysing the  GDP of 170 counttires from 1999 to 2018

## Getting the data

In [1]:
import pandas as pd


## GDP DS
### This dataset contains GDP data for every country from 1999 to 2022 in billions of US dollars. It is taken from https://www.kaggle.com/code/alejopaullier/gdp-by-country-1999-2022/data?scriptVersionId=105241333. This dataset has missing values for some countries, as well as for the whole year 2011. It is not the best choice in sence of informational content of the dataset. But the more problem we have got,  the better this dataset is to practice data cleaning.

In [2]:
gdp = pd.read_csv(r"C:\Users\1\Desktop\GDP by Country 1999-2022.csv", index_col=[0])
gdp

1999     2000     2001    2002    2003    2004  \
Country                                                                   
Afghanistan, Rep. of.       0        0        0   4.084   4.585   5.971   
Albania                 3.444    3.695    4.096   4.456     5.6   7.452   
Algeria                48.845   54.749   55.181  57.053  68.013  85.016   
Angola                  6.153    9.135    8.936  11.386  13.956    19.8   
Antigua and Barbuda     0.652    0.678     0.71   0.718   0.754   0.818   
...                       ...      ...      ...     ...     ...     ...   
Venezuela              97.977  117.153  122.872  92.889  83.442   112.8   
Vietnam                28.684   31.196   32.504  35.148   39.63  45.548   
Yemen, Republic of       7.53    9.561    9.533   9.985  11.869  13.565   
Zambia                  3.132    3.238     3.64   3.775   4.326    5.44   
Zimbabwe                5.963    8.136   12.883  30.856  10.515     4.7   

                          2005     2006     2007     2008  ...     2013  \
Country                                                    ...            
Afghanistan, Rep. of.    7.309    8.399    9.892   11.513  ...   21.555   
Albania                  8.376    9.133   10.163   11.131  ...    14.91   
Algeria                 102.38  114.322  116.158  126.889  ...  190.432   
Angola                  30.632   43.759    55.37   67.608  ...  136.415   
Antigua and Barbuda      0.875    0.962    1.026    1.074  ...    1.404   
...                        ...      ...      ...      ...  ...      ...   
Venezuela              143.443  181.608  219.372  231.959  ...  403.123   
Vietnam                 53.053   60.995   68.298   76.414  ...  135.729   
Yemen, Republic of      15.193     18.7   21.657   24.504  ...   40.003   
Zambia                   7.271   10.942   10.104   10.519  ...   21.829   
Zimbabwe                 4.552     5.54   18.281   63.039  ...    6.214   

                          2014     2015     2016     2017     2018     2019  \
Country                                                                       
Afghanistan, Rep. of.   24.304        0        0        0        0        0   
Albania                 16.053   11.591   12.204   13.214   14.341   15.553   
Algeria                203.449  175.077   181.71  192.256  202.179  210.906   
Angola                 151.089  102.011   98.815  105.369  112.533  119.403   
Antigua and Barbuda      1.494    1.285    1.328    1.386    1.458    1.536   
...                        ...      ...      ...      ...      ...      ...   
Venezuela              409.562  131.855  133.534  144.227  155.096   170.41   
Vietnam                148.914  198.805   214.75  229.845  247.415  265.987   
Yemen, Republic of      42.687        0        0        0        0        0   
Zambia                  23.613   24.466   25.158    27.17   29.911   32.957   
Zimbabwe                 6.851   13.905   14.292   14.994   15.736   16.609   

                          2020     2021     2022  
Country                                           
Afghanistan, Rep. of.        0        0        0  
Albania                 16.996    16.77   18.012  
Algeria                 219.16  163.812  168.195  
Angola                  127.15   70.339   74.953  
Antigua and Barbuda      1.617    1.405    1.534  
...                        ...      ...      ...  
Venezuela              184.364   44.893   43.546  
Vietnam                287.257  368.002  415.493  
Yemen, Republic of           0        0        0  
Zambia                  36.316   21.699   23.967  
Zimbabwe                17.529   25.791   27.806  

[180 rows x 23 columns]

#### Let's drop the missing and zero values

In [3]:
gdp=gdp.dropna()
gdp = gdp.loc[(gdp != "0").all(axis=1)]
gdp.shape

(161, 23)

#### The large numbers are save in DS with comma and dot like this 1, 123. 789. This format can not be converted into float. Let's delete the commas to solve this problem. At the end we will drop the columns after the year 2018, because we don't have this imformation in population DS

In [4]:
for col in gdp.columns:
    gdp[col]=gdp[col].astype(str).str.replace(",","")
    gdp[col]=gdp[col].astype(float)
gdp = gdp.iloc[:,:-4]

## Population DS
### In this DS the values are given in thousands of people for each contry from year 1 to 2018. It is taken from https://www.kaggle.com/datasets/nicklitwinow/population-by-country
### Reduce the population dataset for required years, check if some values are missing. Then set the year column as an index

In [5]:
population = pd.read_csv(r"C:\Users\1\Desktop\Population.csv")

In [6]:
population=population[population["Year"]>1998]
if population.shape==population.dropna().shape:
    print("No values are missing")

No values are missing


In [7]:
population["Year"] = population["Year"].astype(int)
population=population.set_index("Year")
population.head()

Afghanistan     Angola   Albania United Arab Emirates  Argentina  \
Year                                                                    
1999     22,206     11,334     3,193                2,962     36,924    
2000     22,461     11,676     3,164                3,130     37,336    
2001     22,507     12,030     3,134                3,299     37,740    
2002     23,600     12,443     3,104                3,475     38,132    
2003     25,005     12,936     3,075                3,708     38,530    

       Armenia  Australia   Austria Azerbaijan   Burundi  ...   Uruguay  \
Year                                                      ...             
1999    2,970     18,683     8,039      8,056     6,128   ...    3,234    
2000    2,966     18,902     8,059      8,127     6,282   ...    3,246    
2001    2,965     19,144     8,090      8,194     6,466   ...    3,257    
2002    2,962     19,361     8,129      8,257     6,670   ...    3,268    
2003    2,955     19,580     8,169      8,319     6,897   ...    3,277    

     United States Uzbekistan Venezuela (Bolivarian Republic of)   Viet Nam  \
Year                                                                          
1999      279,609     24,701                             23,205     78,170    
2000      282,738     25,042                             23,621     79,207    
2001      285,550     25,372                             24,034     80,239    
2002      288,212     25,696                             24,451     81,288    
2003      290,700     25,992                             24,870     82,327    

          Yemen Former Yugoslavia South Africa     Zambia   Zimbabwe  
Year                                                                  
1999    17,941            22,541       44,320     10,170     11,985   
2000    18,459            22,507       44,882     10,475     12,104   
2001    18,999            22,580       45,451     10,791     12,208   
2002    19,556            22,689       46,073     11,078     12,281   
2003    20,130            22,784       46,738     11,353     12,318   

[5 rows x 169 columns]

#### For some reason there are whitespaces befor and after each number. We need to remove them, as well as commas to convert the str obj type to the float64

In [8]:
for col in population.columns:
    population[col]=population[col].astype(str).str.replace(" ","")
    population[col]=population[col].astype(str).str.replace(",","")
population.iloc[0,1]

'11334'

#### Now we convert every colomn to float

In [9]:
for col in population.columns:
    population[col]=population[col].astype(float)
population.dtypes

Afghanistan             float64
Angola                  float64
Albania                 float64
United Arab Emirates    float64
Argentina               float64
                         ...   
Yemen                   float64
Former Yugoslavia       float64
South Africa            float64
Zambia                  float64
Zimbabwe                float64
Length: 169, dtype: object

#### Check if some values are missing

In [10]:
population.isin([0.0]).any().sum()

0

#### Let's transpose DS and set Country column as index, to make it look like the gdp DS

In [11]:
population = population.T
population.index.name="Country"
population.head()

Year                     1999     2000     2001     2002     2003     2004  \
Country                                                                      
Afghanistan           22206.0  22461.0  22507.0  23600.0  25005.0  25698.0   
Angola                11334.0  11676.0  12030.0  12443.0  12936.0  13480.0   
Albania                3193.0   3164.0   3134.0   3104.0   3075.0   3051.0   
United Arab Emirates   2962.0   3130.0   3299.0   3475.0   3708.0   4064.0   
Argentina             36924.0  37336.0  37740.0  38132.0  38530.0  38946.0   

Year                     2005     2006     2007     2008     2009     2010  \
Country                                                                      
Afghanistan           26335.0  27154.0  27387.0  27706.0  28484.0  29121.0   
Angola                14012.0  14535.0  15083.0  15658.0  16260.0  16884.0   
Albania                3030.0   3011.0   2997.0   2990.0   2988.0   2992.0   
United Arab Emirates   4583.0   5294.0   6161.0   7081.0   7908.0   8540.0   
Argentina             39368.0  39762.0  40135.0  40532.0  40944.0  41359.0   

Year                     2011     2012     2013     2014     2015     2016  \
Country                                                                      
Afghanistan           29758.0  30420.0  31108.0  31823.0  32564.0  33332.0   
Angola                17524.0  18202.0  18907.0  19623.0  20353.0  21097.0   
Albania                3000.0   3008.0   3017.0   3026.0   3035.0   3044.0   
United Arab Emirates   8936.0   9131.0   9187.0   9203.0   9252.0   9350.0   
Argentina             41792.0  42217.0  42636.0  43058.0  43475.0  43887.0   

Year                     2017     2018  
Country                                 
Afghanistan           34125.0  34941.0  
Angola                21858.0  22638.0  
Albania                3053.0   3063.0  
United Arab Emirates   9476.0   9619.0  
Argentina             44294.0  44695.0

## Merge the both DS together 
### Check how much common countries they have with intersection

In [12]:
common=set(gdp.index).intersection(set(population.index))
different=set(gdp.index).symmetric_difference(set(population.index))
#print(set(gdp_final.columns).symmetric_difference(set(population.columns)))
len(common)

129

In [13]:
print(set(gdp.index)-common, set(population.index)-common, sep="\n")

{'Bhutan', 'Papua New Guinea', 'Sudan', 'Fiji', 'St. Lucia', 'Samoa', 'St. Vincent and the Grenadines', 'Maldives', 'Tanzania', 'São Tomé and Príncipe', 'Venezuela', 'Moldova', 'Antigua and Barbuda', 'Hong Kong SAR', 'Belize', 'Russia', 'Suriname', 'Eritrea', 'Slovak Republic', 'St. Kitts and Nevis', 'Kiribati', 'Solomon Islands', 'Kyrgyz Republic', 'Taiwan Province of China', 'Guyana', 'Korea', 'Tonga', 'Bolivia', 'Brunei Darussalam', 'Vietnam', 'Grenada', 'Vanuatu'}
{'Former USSR', 'State of Palestine', 'Yemen', 'Republic of Moldova', 'Sudan (Former)', 'Cuba', 'U.R. of Tanzania: Mainland', 'Syrian Arab Republic', 'Russian Federation', 'Viet Nam', "Lao People's DR", 'Afghanistan', 'Puerto Rico', 'Egypt', 'Saint Lucia', 'China, Hong Kong SAR', 'Taiwan, Province of China', 'Congo', 'TFYR of Macedonia', 'Turkmenistan', 'Cabo Verde', 'D.P.R. of Korea', 'Slovakia', 'Republic of Korea', 'Montenegro', 'Ethiopia', 'Bolivia (Plurinational State of)', 'Venezuela (Bolivarian Republic of)', 'Iran

#### Let's manually rename some columns, so we have more common counrties

In [14]:
population=population.rename(index={"Republic of Korea":"Korea",
                           'Taiwan, Province of China':"Taiwan Province of China",
                           'Russian Federation': 'Russia',
                           'Viet Nam' : 'Vietnam',
                           'Sudan (Former)' : 'Sudan',
                           'Slovakia':'Slovak Republic',
                           'Congo':'D.R. of the Congo',
                           'Republic of Moldova':'Moldova',
                           'Bolivia (Plurinational State of)':'Bolivia',
                           'Kyrgyzstan':'Kyrgyz Republic',
                           'Saint Lucia':'St. Lucia',
                           'U.R. of Tanzania: Mainland':'Tanzania'
                            
                          })
common=set(gdp.index).intersection(set(population.index))
different=set(gdp.index).symmetric_difference(set(population.index))

In [15]:
gdp_merge = gdp.drop(index=list(set(gdp.index)-common))
population_merge = population.drop(index=list(set(population.index)-common))
if population_merge.shape==gdp_merge.shape:
    print("Both DS are equal")
else:
    print("The DS are not equal")

The DS are not equal


In [16]:
print(gdp_merge.shape)
print(population_merge.shape)

(140, 19)
(140, 20)


### Unfortunately the GDP dataset misses the column for 2011. Let us simly take this information from another dataset. So we can avoid cleaning the whole file again. The dataset is taken from: https://www.kaggle.com/datasets/mruanova/gdp2017. It presents the GDP of countries in billions of USD

In [17]:
#There is only one indicator the GDP in USD
year_2011 = pd.read_csv(r"C:\Users\1\Desktop\GDP(only for 2011).csv",index_col="Country Name")
print(year_2011["Indicator Name"].unique())
print(year_2011["Indicator Code"].unique())
year_2011.head()

['GDP (current US$)']
['NY.GDP.MKTP.CD']


Country Code     Indicator Name  Indicator Code         1960  \
Country Name                                                                
Aruba                 ABW  GDP (current US$)  NY.GDP.MKTP.CD          NaN   
Afghanistan           AFG  GDP (current US$)  NY.GDP.MKTP.CD  537777811.1   
Angola                AGO  GDP (current US$)  NY.GDP.MKTP.CD          NaN   
Albania               ALB  GDP (current US$)  NY.GDP.MKTP.CD          NaN   
Andorra               AND  GDP (current US$)  NY.GDP.MKTP.CD          NaN   

                     1961         1962         1963         1964  \
Country Name                                                       
Aruba                 NaN          NaN          NaN          NaN   
Afghanistan   548888895.6  546666677.8  751111191.1  800000044.4   
Angola                NaN          NaN          NaN          NaN   
Albania               NaN          NaN          NaN          NaN   
Andorra               NaN          NaN          NaN          NaN   

                      1965          1966  ...          2008          2009  \
Country Name                              ...                               
Aruba                  NaN           NaN  ...  2.791961e+09  2.498933e+09   
Afghanistan   1.006667e+09  1.400000e+09  ...  1.019053e+10  1.248694e+10   
Angola                 NaN           NaN  ...  8.417804e+10  7.549239e+10   
Albania                NaN           NaN  ...  1.288135e+10  1.204421e+10   
Andorra                NaN           NaN  ...  4.007353e+09  3.660531e+09   

                      2010          2011          2012          2013  \
Country Name                                                           
Aruba         2.467704e+09  2.584464e+09           NaN           NaN   
Afghanistan   1.593680e+10  1.793024e+10  2.053654e+10  2.026425e+10   
Angola        8.252614e+10  1.041160e+11  1.139230e+11  1.249130e+11   
Albania       1.192695e+10  1.289087e+10  1.231978e+10  1.277628e+10   
Andorra       3.355695e+09  3.442063e+09  3.164615e+09  3.281585e+09   

                      2014          2015          2016          2017  
Country Name                                                          
Aruba                  NaN           NaN           NaN           NaN  
Afghanistan   2.061610e+10  1.921556e+10  1.946902e+10  2.081530e+10  
Angola        1.267300e+11  1.026210e+11  9.533720e+10  1.242090e+11  
Albania       1.322824e+10  1.138693e+10  1.188368e+10  1.303935e+10  
Andorra       3.350736e+09  2.811489e+09  2.877312e+09  3.012914e+09  

[5 rows x 61 columns]

##### Taking the date for 2011, devide it by 10**9 , to get the right representation in billions of USD
 

In [18]:
year_2011=year_2011[["2011"]]
year_2011
year_2011 = (year_2011/10**9)
year_2011

2011
Country Name            
Aruba           2.584464
Afghanistan    17.930239
Angola        104.116000
Albania        12.890868
Andorra         3.442063
...                  ...
Kosovo          6.686683
Yemen, Rep.    32.726417
South Africa  416.878000
Zambia         23.460098
Zimbabwe       12.098451

[264 rows x 1 columns]

#### Once angain checking how much common countries we have with intersection of sets 

In [19]:
common=set(year_2011.index).intersection(set(gdp_merge.index))

In [20]:
print(len(common))

136


In [21]:
year_2011=year_2011.rename(index={'Korea, Rep.':"Korea",
                           'Russian Federation': 'Russia',
                           'Viet Nam' : 'Vietnam',
                           'Sudan (Former)' : 'Sudan',
                           'Slovakia':'Slovak Republic',
                           'Congo':'D.R. of the Congo',
                           'Republic of Moldova':'Moldova',
                           'Bolivia (Plurinational State of)':'Bolivia',
                           'Kyrgyzstan':'Kyrgyz Republic',
                           'Saint Lucia':'St. Lucia',
                           'U.R. of Tanzania: Mainland':'Tanzania',
                           "Cote d'Ivoire":"Côte d'Ivoire"
                          })

In [22]:
len(set(year_2011.index).intersection(set(gdp_merge.index)))

139

#### Taking only the intersection countires

In [23]:
year_2011=year_2011.reindex(index=set(year_2011.index).intersection(set(gdp_merge.index)))
year_2011.sort_index()

2011
Country Name            
Albania        12.890868
Algeria       200.019000
Angola        104.116000
Argentina     530.163000
Armenia        10.142111
...                  ...
Uruguay        47.962439
Uzbekistan     45.915191
Vietnam       135.539000
Zambia         23.460098
Zimbabwe       12.098451

[139 rows x 1 columns]

#### gdp_merge still has more countries, but let us just drop the missing values after the concatination to remove countries that have no data for the year 2011 

In [24]:
gdp_merge.shape

(140, 19)

In [25]:
gdp_merge = pd.concat([gdp_merge, year_2011],axis=1)
gdp_merge = gdp_merge.dropna()

In [26]:
gdp_merge

1999     2000     2001    2002     2003     2004     2005  \
Albania       3.444    3.695    4.096   4.456    5.600    7.452    8.376   
Algeria      48.845   54.749   55.181  57.053   68.013   85.016  102.380   
Angola        6.153    9.135    8.936  11.386   13.956   19.800   30.632   
Argentina   283.523  284.204  268.697  97.732  127.643  151.958  181.549   
Armenia       1.845    1.912    2.118   2.376    2.807    3.573    4.903   
...             ...      ...      ...     ...      ...      ...      ...   
Uruguay      20.913   20.086   18.561  12.089   11.211   13.268   16.878   
Uzbekistan   17.041   13.717   11.632   9.657   10.129   12.001   13.670   
Vietnam      28.684   31.196   32.504  35.148   39.630   45.548   53.053   
Zambia        3.132    3.238    3.640   3.775    4.326    5.440    7.271   
Zimbabwe      5.963    8.136   12.883  30.856   10.515    4.700    4.552   

               2006     2007     2008     2009     2010     2012     2013  \
Albania       9.133   10.163   11.131   11.726   11.797   13.808   14.910   
Algeria     114.322  116.158  126.889  134.797  154.843  177.830  190.432   
Angola       43.759   55.370   67.608   69.708   87.718  118.426  136.415   
Argentina   212.702  247.096  277.912  301.331  296.109  315.832  329.332   
Armenia       6.410    7.666    8.579    8.683    8.238    9.086    9.844   
...             ...      ...      ...      ...      ...      ...      ...   
Uruguay      19.221   20.725   22.211   31.606   33.864   47.355   54.388   
Uzbekistan   16.088   18.840   21.727   30.321   34.232   44.892   51.125   
Vietnam      60.995   68.298   76.414   91.764  102.906  123.505  135.729   
Zambia       10.942   10.104   10.519   12.293   15.792   20.208   21.829   
Zimbabwe      5.540   18.281   63.039    3.556    4.220    5.528    6.214   

               2014     2015     2016     2017     2018        2011  
Albania      16.053   11.591   12.204   13.214   14.341   12.890868  
Algeria     203.449  175.077  181.710  192.256  202.179  200.019000  
Angola      151.089  102.011   98.815  105.369  112.533  104.116000  
Argentina   347.731  578.705  578.692  589.585  602.868  530.163000  
Armenia      10.821   10.607   10.672   11.074   11.630   10.142111  
...             ...      ...      ...      ...      ...         ...  
Uruguay      63.167   54.968   57.151   59.943   63.134   47.962439  
Uzbekistan   58.224   65.953   68.699   73.457   78.310   45.915191  
Vietnam     148.914  198.805  214.750  229.845  247.415  135.539000  
Zambia       23.613   24.466   25.158   27.170   29.911   23.460098  
Zimbabwe      6.851   13.905   14.292   14.994   15.736   12.098451  

[139 rows x 20 columns]

In [27]:
gdp_merge=gdp_merge.reindex(columns=sorted(gdp_merge.columns))

### Now we have full data for the GDP and can add population data

In [28]:
population_merge = population_merge.drop(labels=['Taiwan Province of China'])
population_merge.head()
print(gdp_merge.shape)
gdp_merge

(139, 20)


1999     2000     2001    2002     2003     2004     2005  \
Albania       3.444    3.695    4.096   4.456    5.600    7.452    8.376   
Algeria      48.845   54.749   55.181  57.053   68.013   85.016  102.380   
Angola        6.153    9.135    8.936  11.386   13.956   19.800   30.632   
Argentina   283.523  284.204  268.697  97.732  127.643  151.958  181.549   
Armenia       1.845    1.912    2.118   2.376    2.807    3.573    4.903   
...             ...      ...      ...     ...      ...      ...      ...   
Uruguay      20.913   20.086   18.561  12.089   11.211   13.268   16.878   
Uzbekistan   17.041   13.717   11.632   9.657   10.129   12.001   13.670   
Vietnam      28.684   31.196   32.504  35.148   39.630   45.548   53.053   
Zambia        3.132    3.238    3.640   3.775    4.326    5.440    7.271   
Zimbabwe      5.963    8.136   12.883  30.856   10.515    4.700    4.552   

               2006     2007     2008     2009     2010        2011     2012  \
Albania       9.133   10.163   11.131   11.726   11.797   12.890868   13.808   
Algeria     114.322  116.158  126.889  134.797  154.843  200.019000  177.830   
Angola       43.759   55.370   67.608   69.708   87.718  104.116000  118.426   
Argentina   212.702  247.096  277.912  301.331  296.109  530.163000  315.832   
Armenia       6.410    7.666    8.579    8.683    8.238   10.142111    9.086   
...             ...      ...      ...      ...      ...         ...      ...   
Uruguay      19.221   20.725   22.211   31.606   33.864   47.962439   47.355   
Uzbekistan   16.088   18.840   21.727   30.321   34.232   45.915191   44.892   
Vietnam      60.995   68.298   76.414   91.764  102.906  135.539000  123.505   
Zambia       10.942   10.104   10.519   12.293   15.792   23.460098   20.208   
Zimbabwe      5.540   18.281   63.039    3.556    4.220   12.098451    5.528   

               2013     2014     2015     2016     2017     2018  
Albania      14.910   16.053   11.591   12.204   13.214   14.341  
Algeria     190.432  203.449  175.077  181.710  192.256  202.179  
Angola      136.415  151.089  102.011   98.815  105.369  112.533  
Argentina   329.332  347.731  578.705  578.692  589.585  602.868  
Armenia       9.844   10.821   10.607   10.672   11.074   11.630  
...             ...      ...      ...      ...      ...      ...  
Uruguay      54.388   63.167   54.968   57.151   59.943   63.134  
Uzbekistan   51.125   58.224   65.953   68.699   73.457   78.310  
Vietnam     135.729  148.914  198.805  214.750  229.845  247.415  
Zambia       21.829   23.613   24.466   25.158   27.170   29.911  
Zimbabwe      6.214    6.851   13.905   14.292   14.994   15.736  

[139 rows x 20 columns]

#### Let's define the fuction that will melt a dataframe, to produce one column for GDP and Population from two datasets

In [29]:
def melt_and_sort(df,col_name):
    if df.index.name == None:
        df.index.name = "Country"
    df = df.reset_index(level=0)
    df = pd.melt(df, id_vars = "Country", value_vars = df.columns, 
                          var_name = "Year", value_name = col_name)
    df = df.sort_values(["Country","Year"])
    df= df.set_index(["Country"])
    return df
gdp_merge = melt_and_sort(gdp_merge, "GDP_bln_USD")
population_merge= melt_and_sort(population_merge, "Population")

In [30]:
gdp = pd.concat([gdp_merge, population_merge["Population"]], axis=1)


In [31]:
gdp = gdp.reset_index(level=0)
gdp["GDP_bln_USD"]=round(gdp["GDP_bln_USD"],2)

#### Now we have the DS with only four columns. Let's calculate on it the growth rate of GDP, GDP per capita and additionaly growth rate of GDP per capita. We can do it by adding the new columns to gdp dataset. For growth rate we will use the pct_change() method

In [32]:
gdp

Country  Year  GDP_bln_USD  Population
0      Albania  1999         3.44      3193.0
1      Albania  2000         3.70      3164.0
2      Albania  2001         4.10      3134.0
3      Albania  2002         4.46      3104.0
4      Albania  2003         5.60      3075.0
...        ...   ...          ...         ...
2775  Zimbabwe  2014         6.85     13314.0
2776  Zimbabwe  2015        13.90     13479.0
2777  Zimbabwe  2016        14.29     13665.0
2778  Zimbabwe  2017        14.99     13870.0
2779  Zimbabwe  2018        15.74     14097.0

[2780 rows x 4 columns]

In [33]:
gdp["GDP_per_Capita"]=(gdp["GDP_bln_USD"]*(10**9)) / (gdp["Population"]*(10**3))
gdp["GDP_per_Capita"]=gdp["GDP_per_Capita"].astype(int)

In [34]:
gdp.head()

Country  Year  GDP_bln_USD  Population  GDP_per_Capita
0  Albania  1999         3.44      3193.0            1077
1  Albania  2000         3.70      3164.0            1169
2  Albania  2001         4.10      3134.0            1308
3  Albania  2002         4.46      3104.0            1436
4  Albania  2003         5.60      3075.0            1821

In [35]:
gdp["Growth_rate"] = round(gdp["GDP_bln_USD"].pct_change()*100,2)


In [36]:
gdp.head()

Country  Year  GDP_bln_USD  Population  GDP_per_Capita  Growth_rate
0  Albania  1999         3.44      3193.0            1077          NaN
1  Albania  2000         3.70      3164.0            1169         7.56
2  Albania  2001         4.10      3134.0            1308        10.81
3  Albania  2002         4.46      3104.0            1436         8.78
4  Albania  2003         5.60      3075.0            1821        25.56

In [37]:
gdp["growth_per_capita"]=round(gdp["GDP_per_Capita"].pct_change()*100,2)


#### After this calculations let's drop the missing values. We don't have data for 1998, so we can't tell what was the growth in 1999 

In [38]:
gdp=gdp.drop(labels=gdp[gdp["Year"]=='1999'].index)

In [39]:
gdp.query("Year=='1999'")

Empty DataFrame
Columns: [Country, Year, GDP_bln_USD, Population, GDP_per_Capita, Growth_rate, growth_per_capita]
Index: []

#### Now let's reorgonise the columns of the DS 

In [40]:
gdp.columns

Index(['Country', 'Year', 'GDP_bln_USD', 'Population', 'GDP_per_Capita',
       'Growth_rate', 'growth_per_capita'],
      dtype='object')

In [41]:
gdp=gdp[['Country', 'Year', 'Population', 'GDP_bln_USD', 'Growth_rate', 'GDP_per_Capita', 'growth_per_capita']]
gdp

Country  Year  Population  GDP_bln_USD  Growth_rate  GDP_per_Capita  \
1      Albania  2000      3164.0         3.70         7.56            1169   
2      Albania  2001      3134.0         4.10        10.81            1308   
3      Albania  2002      3104.0         4.46         8.78            1436   
4      Albania  2003      3075.0         5.60        25.56            1821   
5      Albania  2004      3051.0         7.45        33.04            2441   
...        ...   ...         ...          ...          ...             ...   
2775  Zimbabwe  2014     13314.0         6.85        10.31             514   
2776  Zimbabwe  2015     13479.0        13.90       102.92            1031   
2777  Zimbabwe  2016     13665.0        14.29         2.81            1045   
2778  Zimbabwe  2017     13870.0        14.99         4.90            1080   
2779  Zimbabwe  2018     14097.0        15.74         5.00            1116   

      growth_per_capita  
1                  8.54  
2                 11.89  
3                  9.79  
4                 26.81  
5                 34.05  
...                 ...  
2775               9.13  
2776             100.58  
2777               1.36  
2778               3.35  
2779               3.33  

[2641 rows x 7 columns]

#### Saving the data 

In [42]:
#gdp.to_csv(r"C:\Users\1\Desktop\GDP DS\GDP_final.csv")